site/pages/home/newsletter-slice/test.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import NewsLetter from '.';
+import AfterSignUp from './after-sign-up';
+import BeforeSignUp from './before-sign-up';
+
+describe('Newsletter slice', () => {
+  it('shows pre-signup content if the form has not been submitted yet - default state', () => {
+    const wrapper = shallow(<NewsLetter />);
+    expect(wrapper.find(BeforeSignUp)).to.have.length(1);
+  });
+
+  it('changes "submitted" state when submitted', () => {
+    const wrapper = shallow(<NewsLetter />);
+    wrapper.instance().onSubmit({
+      preventDefault: () => {},
+    });
+    expect(wrapper.state('newsletterSubmitted')).to.equal(true);
```

Comment:
This test should verify that the handler has been called with the correct data.

Ideally we would not remove the handler and would inject the API client dependency and test that was called as then the handling logic (what the component is responsible for) will be tested.

The client logic needs to be tested too, though outside the context of the component.